source/AI.cpp

```diff
@@ -1256,7 +1258,12 @@ bool AI::FollowOrders(Ship &ship, Command &command) const
 		// way to reach that system (via wormhole or jumping). This may
 		// result in the ship landing to refuel.
 		SelectRoute(ship, it->second.targetSystem);
-		return false;
+		
+		// Travel there even if your parent is not planning to travel.
+		if(ship.GetTargetSystem())
+			MoveIndependent(ship, command);
```

Comment:
I just did some testing (in Shaula) and the ships I order to move appear to prioritize finishing fighting enemies in the current system before they start moving to the desired destination in the system that I ordered them to go to.

If I give an order to a ship to move somewhere, then I would expect the move to be executed immediately, not after all enemies are gone.

Reply:
That's an unfortunate side effect of how player-given orders work, and how we currently tell AI ships to act like an escort (i.e. pay attention to your parent and be near them) vs make their own decisions (e.g. fighting a target, following player-given orders, or performing intersystem travel operations without regard to a parent ship).

Refactoring that sequence of the AI class is something that could be done here, but could also be done in a separate PR. As @tux2603 mentions, making your escorts promptly ignore combat to jump away further expands the "combat only happens when the player can see it" exploit.

Reply:
I'm okay with leaving this for a separate PR (and for leaving it until after fixing the "combat only happens when the player can see it" exploit).